lint381/c.py

```diff
@@ -11,6 +11,20 @@
 
 
 @linter.register
+@with_matched_tokens(start=match_regex("^sizeof$"), end=match_regex("^char$"))
+def prohibited_unsigned_char(source, *, match):
+    """Flag sizeof(char) as it defined to be 1 and is redundant."""
+    error_message = "sizeof(char) should not be used as it is defined by the "\
+                    "C standard to be equal to 1"
+
+    if match[1].value == "char":
+        yield Error(error_message, tokens=[match[0], match[1]])
+    elif (match[2].value == "char" and match[1].value == "("):
```

Comment:
Why would you check `[2]` and then `[1]`? It seems out of order.

This matches `char[x]`, `char*`, and `char (*)()`, and at least one of those might be reasonable.

I'd suggest you add a `lookahead=1` parameter to `with_matched_tokens` to peek at the next character and see if it's a closing brace.